components/exporter/multiexporterexporter/exporter_test.go

```diff
@@ -0,0 +1,171 @@
+			data:     plog.NewLogs(),
+		},
+		{
+			name: "invalid payload for the logs exporter",
+			context: func() context.Context {
+				return collectorclient.NewContext(context.Background(), collectorclient.Info{
+					Metadata: collectorclient.NewMetadata(
+						map[string][]string{mappingKey: {mappingKeyValue}},
+					),
+				})
+			},
+			exporter: logsExporter,
+			data:     ptrace.NewTraces(),
+			error:    "invalid data type \"ptrace.Traces\" for the logs exporter",
+		},
+		{
+			name: "export metrics",
+			context: func() context.Context {
+				return collectorclient.NewContext(context.Background(), collectorclient.Info{
+					Metadata: collectorclient.NewMetadata(
+						map[string][]string{mappingKey: {mappingKeyValue}},
+					),
+				})
+			},
+			exporter: metricsExporter,
+			data:     pmetric.NewMetrics(),
+		},
+		{
+			name: "invalid payload for the metrics exporter",
+			context: func() context.Context {
+				return collectorclient.NewContext(context.Background(), collectorclient.Info{
+					Metadata: collectorclient.NewMetadata(
+						map[string][]string{mappingKey: {mappingKeyValue}},
+					),
+				})
+			},
+			exporter: metricsExporter,
+			data:     ptrace.NewTraces(),
+			error:    "invalid data type \"ptrace.Traces\" for the metrics exporter",
+		},
+		{
+			name: "export traces",
+			context: func() context.Context {
+				return collectorclient.NewContext(context.Background(), collectorclient.Info{
+					Metadata: collectorclient.NewMetadata(
+						map[string][]string{mappingKey: {mappingKeyValue}},
+					),
+				})
+			},
+			exporter: tracesExporter,
+			data:     ptrace.NewTraces(),
+		},
+		{
+			name: "invalid payload for the traces exporter",
+			context: func() context.Context {
+				return collectorclient.NewContext(context.Background(), collectorclient.Info{
+					Metadata: collectorclient.NewMetadata(
+						map[string][]string{mappingKey: {mappingKeyValue}},
+					),
+				})
+			},
+			exporter: tracesExporter,
+			data:     plog.NewLogs(),
+			error:    "invalid data type \"plog.Logs\" for the traces traces",
+		},
+		{
+			name: "missing exporter",
+			context: func() context.Context {
+				return collectorclient.NewContext(context.Background(), collectorclient.Info{
+					Metadata: collectorclient.NewMetadata(
+						map[string][]string{mappingKey: {"value1"}},
+					),
+				})
+			},
+			exporter: logsExporter,
+			data:     plog.NewLogs(),
+			error:    "exporter is not found for the mapping key: \"value1\"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			e := newMultiExporter(
+				map[string]interface{}{
+					mappingKeyValue: tt.exporter,
+				},
+				mappingKey,
+				zap.NewNop(),
+			)
+
+			err := e.consume(tt.context(), tt.data)
+			if tt.error != "" {
+				assert.ErrorContains(t, err, tt.error)
+				return
+			}
+		})
+	}
+}
+
+func TestRetrieveFromContext(t *testing.T) {
+	md := collectorclient.NewMetadata(map[string][]string{
+		"X-Scope-Orgid": {"123"},
```

Comment:
Can we cross-test this with the case normalization? Like, use `OrgID` here and `Orgid` on line 168 (leaving a comment stating that this is on purpose)